src/Helpers/var.php

```diff
@@ -9,68 +9,112 @@
+ * @throws LangException
+ * @throws LoggerException
  */
-function error($var, ?ReportableInterface $loggerAdapter = null)
+function error($var)
 {
-    $logger = new Logger($loggerAdapter ?: new MessageLogger);
-    $logger->error($var);
+    $logger = LoggerManager::getHandler();
+
+    if (is_debug_mode() || LoggerConfig::getLogLevel(LogLevel::ERROR) >= LoggerConfig::getAppLogLevel()) {
+        $logger->error($var);
+    }
 }
 
 /**
  * Reports warning
  * @param mixed $var
- * @param ReportableInterface|null $loggerAdapter
+ * @return void
+ * @throws ConfigException
+ * @throws DiException
+ * @throws LangException
+ * @throws LoggerException
+ * @throws ReflectionException
  */
-function warning($var, ?ReportableInterface $loggerAdapter = null)
+function warning($var)
 {
-    $logger = new Logger($loggerAdapter ?: new MessageLogger);
-    $logger->warning($var);
+    $logger = LoggerManager::getHandler();
+
+    if (is_debug_mode() || LoggerConfig::getLogLevel(LogLevel::WARNING) >= LoggerConfig::getAppLogLevel()) {
+        $logger->warning($var);
+    }
 }
 
 /**
  * Reports notice
  * @param mixed $var
- * @param ReportableInterface|null $loggerAdapter
+ * @return void
+ * @throws ConfigException
+ * @throws DiException
+ * @throws LangException
+ * @throws LoggerException
+ * @throws ReflectionException
  */
-function notice($var, ?ReportableInterface $loggerAdapter = null)
+function notice($var)
 {
-    $logger = new Logger($loggerAdapter ?: new MessageLogger);
-    $logger->notice($var);
+    $logger = LoggerManager::getHandler();
+
+    if (is_debug_mode() || LoggerConfig::getLogLevel(LogLevel::NOTICE) >= LoggerConfig::getAppLogLevel()) {
+        $logger->notice($var);
+    }
 }
 
 /**
  * Reports info
  * @param mixed $var
- * @param ReportableInterface|null $loggerAdapter
+ * @return void
+ * @throws ConfigException
+ * @throws DiException
+ * @throws LangException
+ * @throws LoggerException
+ * @throws ReflectionException
  */
-function info($var, ?ReportableInterface $loggerAdapter = null)
+function info($var)
 {
-    $logger = new Logger($loggerAdapter ?: new MessageLogger);
-    $logger->info($var);
+    $logger = LoggerManager::getHandler();
+
+    if (is_debug_mode() || LoggerConfig::getLogLevel(LogLevel::INFO) >= LoggerConfig::getAppLogLevel()) {
+        $logger->info($var);
+    }
 }
 
 /**
  * Reports debug
  * @param mixed $var
- * @param ReportableInterface|null $loggerAdapter
+ * @return void
+ * @throws ConfigException
+ * @throws DiException
+ * @throws LangException
+ * @throws LoggerException
+ * @throws ReflectionException
  */
-function debug($var, ?ReportableInterface $loggerAdapter = null)
+function debug($var)
 {
-    $logger = new Logger($loggerAdapter ?: new MessageLogger);
-    $logger->debug($var);
+    $logger = LoggerManager::getHandler();
+
+    if (is_debug_mode() || LoggerConfig::getLogLevel(LogLevel::DEBUG) >= LoggerConfig::getAppLogLevel()) {
+        $logger->debug($var);
+    }
 }
 
```

Comment:
The methods above are very similar. If we can reduce the code duplication, we should do so.